### PR TITLE
#85zrrxhf3 → Update user from user management hook

### PR DESCRIPTION
**Pull request by Miguel-Invorious**
Created functions `update<T>(data:Partial<T>):Promise<T>`, `getProfile<T>():Promise<T>` , to update or get user profile information.

```typescript
export function useUserManagement(
  config?: CreateAxiosDefaults,
): IUserManagement {
  const { post, requestError, get, put } = useHttpClient({
    config,
  });
  async function register<T>(data: Partial<T>) {
    return await post<T>('/register', {
      ...data,
    });
  }

  async function getProfile<T>() {
    return await get<T>('/me');
  }

  async function update<T>(data: Partial<T>) {
    return await put<T>('/me', {
      ...data,
    });
  }
  return {
    register,
    requestError,
    getProfile,
    update,
  };
}
```

To implement getProfile function we can do it inside the useEffect hook

```typescript
useEffect(() => {
  const fetchUser = async () => {
    const user = await getProfile<IUser>();
    if (!requestError) {
      setUser(user);
    }
  };
  setFetchingUser(true);
  fetchUser();
  setFetchingUser(false);
}, []);
```

And the update function inside a form, to be called once its submitted by the user.

```typescript
const handleUpdate = async (event: React.FormEvent<HTMLFormElement>) => {
  event.preventDefault();
  setFormData({} as IUser);
  setUpdateProfile(false);
  const updatedUser = await update<IUser>(formData);
  setUser(updatedUser);
};
```

### CODE REVIEW

**Review by Jesus-Rojas — APPROVED** [review]

File: packages/example-front/src/app/modules/profile/profile.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import styles from './profile.module.scss';
+
+export function Profile() {
+  const [updateProfile, setUpdateProfile] = useState(false);
+  const [fetchingUser, setFetchingUser] = useState(false);
+  const [user, setUser] = useState<IUser>();
+  const { requestError, getProfile, update } = useUserManagement({
+    baseURL: 'api/auth',
+    headers: {
+      Authorization: `Bearer ${localStorage.getItem('token')}`,
```
Comment (wialejo-invorious) [inline packages/example-front/src/app/modules/profile/profile.tsx:14]: I think we create a Auth Helper or services in the front to manage this Headers. we will need to add this token it multiple components...

Comment (wialejo-invorious) [reply packages/example-front/src/app/modules/profile/profile.tsx:14]: @Miguel-Invorious i think useUserManagement hook should contain the URL of the API, it is his responsability to manage the user data including where it comes from. thoughts???

Comment (Miguel-Invorious) [reply packages/example-front/src/app/modules/profile/profile.tsx:14]: Yeah I think it should do as well, and I'll create a hook to return auth token, something like useBearearToken()

**Review by Jesus-Rojas — APPROVED** [review]

**Comment by Jesus-Rojas** [comment]
Solved problems in merge from main

**Review by Andres-Invorious — APPROVED** [review]

**Pull request merged**